src/readers/reader_tum.py

```diff
@@ -0,0 +1,101 @@
+import open3d as o3d
+import mrob
+import numpy as np
+import os
+import cv2
+
+from reader import Reader
+from nptyping import NDArray, Shape, Float, UInt8
+from pathlib import Path
+from src.database import Database
+
+
+class ReaderTUM(Reader):
+    def __init__(self, path_to_dataset: str, max_difference: int = 1000):
+        self.path_to_dataset = path_to_dataset
+        self.max_difference = max_difference
+
+    def read_dataset(self) -> Database:
+        return super().read_dataset()
+
+    def _get_images_pcds_traj(
+        self,
+    ) -> tuple[list[str], list[str], list[NDArray[Shape["4, 4"], Float]]]:
+        depth_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "depth")
+        )
+        rgb_with_timestamps = self.__read_folder(
+            os.path.join(self.path_to_dataset, "rgb")
+        )
+        matches = self.__associate(rgb_with_timestamps, depth_with_timestamps)
+        timestamps, traj = self.__read_trajectory()
+        timestamps = np.asarray(timestamps)
+        images_paths = []
+        pcds_paths = []
+        res_traj = []
+        for (a, b) in matches:
+            diff = abs(timestamps - np.mean([a, b]))
+            min_diff_ind = np.argmin(diff)
+            images_paths.append(rgb_with_timestamps[a])
+            pcds_paths.append(depth_with_timestamps[b])
```

Comment:
wtf, it is depth, not pcd

Reply:
it is pointcloud in format of depth image, is not it?